Chat event hook guide

howawong opened the pull request:
Add Chat cloud function event hook guides in Python and JS

Connects #166

howawong committed:
Chat event hook guide in Python

howawong committed:
Chat event hook guide in JS

howawong committed:
Add Introduction to Cloud Functions in link Chat event hook guide

howawong committed:
Typo

howawong committed:
mdspell fix

carmenlau commented:
Content looks good to me. Pending for verifying the code work in skygear code.
1. I think there will be error in `npm install` that we need adding `skygear` in package.json
2. Please also remember to update the demo repo if code update is needed.

Thanks @howawong 🙏🏻

rickmak commented on chat/cloud-event-hook/python.md:
```diff
@@ -0,0 +1,173 @@
+    message = deserialize_record(message)
+    conversation = deserialize_record(conversation)
+    participants = [deserialize_record(p) for p in participants]
+    other_user_ids = [p.id.key for p in participants if p != current_user_id()]
```
Please avoid one-liner

howawong committed:
Minor Update

- Update package.json and demo cloud code require statement in JS
- Update import statement, one line loop and  payload in demo cloud code

howawong commented:
@carmenlau  yes, correct 🙈

carmenlau commented:
@howawong Would like to know the status of js example 🙈

howawong committed:
Update skygear-chat version to 1.3.4 and js example cloud code

howawong committed:
Remove trailing whitespace

howawong commented:
Both Guide PR and cloud code JS PR are updated.

carmenlau commented:
Per discussion, we will use the message user id instead of the context current user in the demo. Please update @howawong, thanks!

howawong committed:
Update currentUser statement

carmenlau commented:
Please add update skygear-doc for the sidebar menu
